plutus-core/plutus-core/src/PlutusCore/Constant/Dynamic/Emit.hs

```diff
@@ -46,3 +46,6 @@ newtype NoEmitterT m a = NoEmitterT
 
 instance Monad m => MonadEmitter (NoEmitterT m) where
     emit _ = pure ()
+
+instance (Monad m, MonadEmitter m) => MonadEmitter (ExceptT e m) where
```

Comment:
The `Monad m` constraint should be redundant? If not, then we should add it as a superclass to `MonadEmitter`.